ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/coordinator/BasicAction.java

```diff
@@ -3377,6 +3382,37 @@ private final boolean checkIsCurrent ()
         return isCurrent;
     }
 
+    // called by the reaper system via TwoPhaseCoordinator.recordStackTraces
+    protected synchronized void recordStackTraces() {
+        Map<String,String> currentCapture = createStackTraces();
+        StackTraceCapture stackTraceCapture = new StackTraceCapture(System.currentTimeMillis(), currentCapture);
+        threadStackTraceHistoryList.add(stackTraceCapture);
+    }
+
+    // called by the reaper system via TwoPhaseCoordinator.outputCapturedStackTraces
+    protected synchronized void outputCapturedStackTraces() {
+        DateFormat dateFormat = new SimpleDateFormat("HH:mm:ss.SSS");
+        for(StackTraceCapture stackTraceCapture : threadStackTraceHistoryList) {
+            for(Map.Entry<String,String> entry : createStackTraces().entrySet()) {
```

Comment:
Please correct me if I am wrong here, I think in the `for` loop we should use the recorded value i.e using `stackTraceCapture.getStackTraces()` instead of `createStackTraces()`?

Reply:
D'oh, yes, that's a bug for sure. Good catch, thanks!